pkg/trace/api/otlp.go

```diff
@@ -239,7 +239,7 @@ func (o *OTLPReceiver) ReceiveResourceSpans(rspans ptrace.ResourceSpans, header
 		containerID = rattr[string(semconv.AttributeK8SPodUID)]
 	}
 	if containerID == "" {
-		containerID = fastHeaderGet(header, headerContainerID)
+		containerID = getContainerID(req)
```

Comment:
Added to the http.Server here. I'm going to skip adding to the gRPC one for now since:
1. It will require a slightly different approach, so additional code
2. It would require UDS support to completely test, which we don't have.
3. It wouldn't be available for users today anyway unless we publish OTLP UDS support, which is beyond the scope of this change.

I think it will be better to add this when we are actually going to support the UDS feature for OTLP and can test it.